adaptors/backbone/base_model.js

```diff
@@ -435,6 +435,66 @@ BaseModel.prototype.bindExposedEvent = function(event, prop, childComponent) {
   });
 };
 
+BaseModel.prototype.validateProperty = function (propTypeDesc, propValue) {
+  var validDescAttrs = ['required', 'type'];
+  var propValueType = typeof propValue;
+  var isRequired = propTypeDesc.required;
+  var type = propTypeDesc.type;
+  var typeValidators = {
+    string: (value) => typeof value === 'string',
+    number: (value) => typeof value === 'number',
+    boolean: (value) => typeof value === 'boolean',
+    object: (value) => typeof value === 'object',
+    custom: (value) => typeof value === 'function',
+    symbol: (value) => typeof value === 'symbol',
+    array: (value) => _.isArray(value)
+  };
+
+  /**
+   * Checks if target object has at least one attribute from attrs array.
+   *
+   * @param  {Object}   target  Object to check for attributes
+   * @param  {Array}    attrs   Array of attributes
+   *
+   * @return {Boolean}
+   */
+  function hasAtLeastOneKey(target, attrs) {
+    var attrsLen = attrs.length;
+    var i = 0;
+    var counter = 0;
+    for (; i < attrsLen; i++) {
+      if (target[attrs[i]] !== undefined) {
+        counter++;
+      }
+    }
+    return !!counter;
+  }
+
+  // check if propTypeDesc is an object and has at lest one valid attribute
+  if (typeof propTypeDesc !== 'object' || !hasAtLeastOneKey(propTypeDesc, validDescAttrs)) {
+    throw({msg: 'invalid property descriptor.'});
+  }
+
+  // validate property type only when it was specified and is a string
+  if (typeof type !== 'string' || Object.keys(typeValidators).indexOf(type) === -1 ) {
```

Comment:
Instead of `Object.keys(typeValidators).indexOf(type)` can we check for `typeof typeValidators[keys] ==='function'`?